### DISCUSSION TITLE
Does the Background Modes entitlement need to be added for an iOS app to use Firebase Cloud Messaging, RemoteConfig, etc.?

### DISCUSSION THREAD

**Discussion by 1oo7** [firebase/firebase-ios-sdk#8977] [Q&A]
Does the Background Modes entitlement need to be added for an iOS app to use Firebase Cloud Messaging, RemoteConfig, etc.?

[Google's official Firebase iOS SDK documentation](https://firebase.google.com/docs/cloud-messaging/ios/client) only says we need to add the Push Notifications entitlement.

However [the React Native Firebase website](https://rnfirebase.io/messaging/usage/ios-setup) says:

> Next the "Background Modes" capability needs to be enabled, along with both the "Background fetch" and "Remote notifications" sub-modes.

Which of these two is correct?

I'm asking, because as of Xcode 13, our app is getting killed during launch on iOS 14 devices where the App Store version of the app gets installed via an MDM Install App Command (but not if MDM installs an ad-hoc build, nor if the user directly downloads and installs the app themselves from the app store). The crash report says "unknown thread" and does not provide a stack trace. 

Just trying to make sure whether we might need some additional entitlements for Crashlytics that, if we didn't have them, might cause some problems...

For details see Apple Developer forum post here: https://developer.apple.com/forums/thread/693606

**Comment by paulb777** [comment]
Based on your description, this may be an issue related to MDM Installs.

I suspect if it were a general issue, it would have come up well before now since Xcode 13 has now been used for several months.

**Reply by 1oo7** [reply]
Yeah but how many enterprise apps are building with a minimum deployment target of iOS 14 or higher, like we are? 

I suspect most enterprise apps run a deployment target of iOS 13 or lower. Which means that dyld won't apply as strict of rules to library loading and a few other details. 

That, combined with something odd about how App Store installs and signs apps when they're distributed over MDM, seems related. 

But it sounds like you're implying that Background Modes entitlement doesn't need to be added. Is that what you're implying?

**Comment by paulb777** [comment]
@mikehardy Do you know why the React Native docs say the modes need to be added?

**Reply by mikehardy** [reply]
That's a good question @paulb777 - unfortunately before my time on the project. I just thought that was the way it is, and I assumed it was carried over from upstream docs. I think I'll have to defer to @Salakar on this one and I'll reach out to him to see if he can shed some light...

**Reply by Salakar** [reply]
I think this is leftovers from when we had a background modes used for advanced local notifications - this feature has since been removed from React Native - so I think the docs should be updated to reflect this now

**Reply by mikehardy** [reply]
queued https://github.com/invertase/react-native-firebase/issues/5934
